proto/query.proto

```diff
@@ -32,20 +32,21 @@ message Query {
         message Ok {
             oneof ok {
                 Empty empty = 1;
-                ReadableConceptTreeStream readable_concept_tree_stream = 3;
+                ConceptDocumentStream concept_document_stream = 3;
                 ConceptRowStream concept_row_stream = 4;
             }
 
-            message Empty {
+            message Empty {}
 
-            }
-
-            message ReadableConceptTreeStream {
+            message ConceptDocumentStream {
                 // note: we could use this first response to record debug info, type annotations, warnings, etc
-                // TODO: Add Type query_type
+                // TODO: network optimisation: replace keys with IDs, sending keys in the header to rebuild the document on the client side
+                // TODO: network optimisation: replace types (== mostly constant strings) with a IDs, sending types in the header to rebuild on the client side
```

Comment:
These are future work , not until benchmarking shows the need to squeeze the network.